control/control_loop.go

```diff
@@ -306,12 +303,32 @@ func (l *Loop) GetConfig(ctx context.Context) Config {
 	return l.cfg
 }
 
-// GetTuning returns the current tuning value.
-func (l *Loop) GetTuning(ctx context.Context) bool {
-	return l.tuning
+// MonitorTuning waits for tuning to start, and then returns once it's done.
+func (l *Loop) MonitorTuning(ctx context.Context) error {
+	// wait until tuning has started
+	for {
+		tuning := l.GetTuning(ctx)
+		if tuning {
+			break
+		}
+	}
```

Comment:
this is totally async from the constructor, correct?

Reply:
no, reconfigure doesn't return until the tuning process is done. I'll try to change that right now, good catch

Reply:
fixed